homeassistant/components/tplink/README.md

```diff
@@ -0,0 +1,31 @@
+# TPLink Integration
+
+This document covers details that new contributors may find helpful when getting started.
+
+## Modules vs Features
+
+The kasa library which this integration depends on exposes functionality via modules and features.
+The `Module` apis encapsulate groups of functionality provided by a device,
+e.g. Light which has multiple attributes and methods such as `set_has`, `brightness` etc.
+The `features` encapsulate unitary functions and allow for introspection.
+e.g. `on_since`, `voltage` etc.
+
+If the integration implements a platform that presents single functions or data points, such as `sensor`,
+`button`, `switch` it uses features.
+If it's implementing a platform with more complex functionality like `light`, `fan` or `climate` it will
+use modules.
+
+## Dynamic feature creation
+
+Entities can be dynamically created by inspecting various properties of a feature such
+as `type`, `unit` or `precision`. If more data is needed than the feature provides,
```

Comment:
```suggestion
as `type`, `unit` or `precision`.
If more data is needed than the feature provides,
```

Reply:
Now done in https://github.com/home-assistant/core/pull/120025